x-pack/packages/ml/aiops_log_rate_analysis/queries/mini_histogram_utils.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import type { NumericChartData, SignificantItemHistogramItem } from '@kbn/ml-agg-utils';
+
+import type { AiopsLogRateAnalysisSchema } from '../api/schema';
+
+export interface MiniHistogramAgg extends estypes.AggregationsSingleBucketAggregateBase {
+  doc_count: number;
+  mini_histogram: {
+    buckets: Array<
+      estypes.AggregationsSingleBucketAggregateBase & estypes.AggregationsHistogramBucketKeys
+    >;
+  };
+}
+
+export const HISTOGRAM_AGG_PREFIX = 'histogram_';
+
+export const getMiniHistogramAgg = (params: AiopsLogRateAnalysisSchema) => {
+  return {
+    mini_histogram: {
+      histogram: {
+        field: params.timeFieldName,
+        interval: (params.end - params.start) / 19,
```

Comment:
Where does `19` come from? Is it the number of bars in the mini histograms?

Reply:
I moved this to a const in [3bc7f6e](https://github.com/elastic/kibana/pull/189155/commits/3bc7f6e98852068e7ef2761274dece673e17b5a8). We want 20 bars, but for calculating the interval we need to do `20-1`.